app/code/core/Mage/Adminhtml/Model/Config.php

```diff
@@ -80,16 +89,67 @@ public function getTabs()
         return $this->_tabs;
     }
 
+    /**
+     * @param array $params
+     */
+    public function __construct(array $params = array())
+    {
+        $this->_cacheChecksum = null;
+        $this->setCache(Mage::app()->getCache());
+        $this->setCacheTags([Mage_Core_Model_Config::CACHE_TAG]);
+        $usesCache = Mage::app()->useCache('config');
+        if (!$usesCache || !$this->loadCache()) {
+            /** @var Mage_Core_Model_Config_Base $config */
+            $this->_config = Mage::getConfig()->loadModulesConfiguration('system.xml')
+                ->applyExtends();
+            if ($usesCache) {
+                $this->saveCache();
+            }
+        }
+    }
+
+    /**
+     * @param $tags
+     * @return $this|Mage_Adminhtml_Model_Config
+     */
+    public function saveCache($tags=null)
+    {
+        if ($this->getCacheSaved()) {
+            return $this;
+        }
+        if (is_null($tags)) {
+            $tags = $this->_cacheTags;
+        }
+        $xmlString = $this->_config->getXmlString();
+        $this->_saveCache($xmlString, $this->getCacheId(), $tags, $this->getCacheLifetime());
+        $this->setCacheSaved(true);
+        return $this;
+    }
+
+    /**
+     * @return bool
+     */
+    public function loadCache()
+    {
+        $xmlString = $this->_loadCache($this->getCacheId());
+        $class = Mage::getConfig()->getModelClassName('core/config_base');
+        $this->_config = new $class();
+        libxml_use_internal_errors(true);
+        if ($this->_config->loadString($xmlString)) {
```

Comment:
Like @luigifab mentioned, the call to `loadString` fails when the cache is empty: `"$string" parameter for simplexml_load_string is not a string`. You'll have to handle that here and load an empty config xml string otherwise (`<?xml version="1.0"?><config></config>`).

Reply:
or with `!empty() && ...` ?

Reply:
@lamskoy could you fix this last comment so that we can approve the PR? Thank you!

Reply:
> @lamskoy could you fix this last comment so that we can approve the PR? Thank you!
A bit later. Have war in my country - Ukraine

Reply:
@lamskoy sorry, I didn't know.

Reply:
I added a check for `$xmlString`.